tracer/src/Datadog.Trace/ExtensionMethods/SpanExtensions.cs

```diff
@@ -96,7 +96,7 @@ internal static void SetHeaderTags<T>(this Span span, T headers, IDictionary<str
             }
         }
 
-        internal static void SetHttpStatusCode(this Span span, int statusCode, bool isServer)
+        internal static void SetHttpStatusCode(this Span span, int statusCode, bool isServer, Tracer tracer)
```

Comment:
Is there any case when this new argument won't be `Tracer.Instance` or have the same settings?

Reply:
Yes, in our tests. This was causing the global Tracer to be swapped out and disposed etc mid-trace

Reply:
I'm not a fan of passing the entire Tracer object here, since we have a chance to refactor the method. Can we either pass in the settings object or a delegate instead?

Reply:
Well the span has already a relationship to the tracer (span -> SpanContext -> TraceContext -> Tracer) so in theory we should use that, but the `ITraceContext` interface is getting on the way

Reply:
Yeah, kind of annoying this isn't directly available from ITraceContext - once we have immutable settings, maybe they could/should be exposed there? 🤔 

In the mean time, updated to pass in `TracerSettings` for now in https://github.com/DataDog/dd-trace-dotnet/pull/1990/commits/e5d7cbf0ce99801c23b07d0570986623a5463cb2